api/client/samples/similar_regions_Frechet/sr.py

```diff
@@ -0,0 +1,560 @@
+                              if self.region_info.get(r,{'level':-1})['level']==level+1]
+                ri.update({r:self.region_info[r] for r in to_include})
+                level += 1
+            self.region_info = ri
+            self._logger.info("Trimmed region info to {} items".format(len(self.region_info)))
+        self.needed_regions = sorted(self.region_info.keys())
+        self.needed_properties = sorted(self.metric_instance.keys())
+        
+        ##################### Data download ################################################
+        self.data = np.zeros((len(self.needed_regions)*t_int_per_year, len(self.metric_instance)))
+        self.data[:] = np.nan
+        for (prop_i, prop_name) in enumerate(self.needed_properties):
+            self.prop_data = self.data[:,prop_i] # alias to specific column of the data matrix
+            prop_path = os.path.join(self.data_dir, prop_name+'_'+str(t_int_per_year)+'.npz')
+            missing_regions = self.needed_regions
+            prop_data_cached = np.array(())
+            prop_regions = []
+            if update != 'reload':
+                if os.path.isfile(prop_path):                    
+                    self._logger.info("Reading property {} from cache {} ...".format(prop_name, prop_path))    
+                    with np.load(prop_path,allow_pickle=True) as prop_file:
+                        prop_data_cached = prop_file['data']
+                        prop_regions = list(prop_file['regions'])
+                    self._logger.info("Merging in cached info ...")
+                    self._merge_into_prop_data(prop_data_cached,prop_regions)
+                    self._logger.info("done")
+                missing_regions = sorted(set(self.needed_regions) - set(prop_regions))
+                if update == 'no':
+                    self._logger.warning("Property {} is missing {} regions out of desired {} and no updates requested".
+                                         format(prop_name, len(missing_regions),len(self.needed_regions)))
+                    missing_regions = []
+                    
+                    
+            # actual download
+            if missing_regions:
+                valid_data, valid_regions = self._load_property_for_regions(prop_name, missing_regions)
+                if valid_regions:
+                    np.savez_compressed(prop_path,
+                                data = np.concatenate([prop_data_cached,valid_data],axis=0),
+                                regions = prop_regions+valid_regions)
+                    self._merge_into_prop_data(valid_data, valid_regions)
+                    
+        ################### Drop incomplete regions #################################
+        self.data = pd.DataFrame(self.data,
+                            index=pd.MultiIndex.from_product([self.needed_regions,range(1,self.t_int_per_year+1)],
+                                                             names=['region','month']),
```

Comment:
instead of 'month' maybe 'year_period'?

Reply:
self.data exists as dataframe for a short time (goes back to numpy array quickly), but sounds good, changed.